src/server/cluster/cluster_family.cc

```diff
@@ -413,19 +436,35 @@ void ClusterFamily::DflyClusterConfig(CmdArgList args, ConnectionContext* cntx)
     return rb->SendError("Invalid JSON cluster config", kSyntaxErrType);
   }
 
-  auto deleted_slot_ids = cluster_config_->SetConfig(json.value());
-  if (!deleted_slot_ids.has_value()) {
+  bool is_first_config = !tl_cluster_config->IsConfigured();
+  SlotSet before = tl_cluster_config->GetOwnedSlots();
+
+  atomic_size_t succeeded = 0;
+  auto cb = [&](util::ProactorBase* pb) {
+    if (tl_cluster_config->SetConfig(json.value())) {
+      ++succeeded;
+    }
+  };
+  server_family_->service().proactor_pool().AwaitFiberOnAll(std::move(cb));
+
+  if (succeeded == 0) {
```

Comment:
what if succeeded < proactor pool size

Reply:
That is a **very** good question. That got me to do a very elegant refactor (imho).
Originally, I thought that a `DCHECK()` (which was below your comment) was sufficient. After all, if `SetConfig()` succeeded in 1 thread, it _should_ succeed in all others. But it's not an ideal solution.
Instead, I opted to change the code so that:
1. Validation is done in a function that _creates_ a new object, so that there is no such thing as an un-configured object anymore
2. Now `isConfigured()` and `SetConfig()` are removed, as they are no longer necessary
3. Instead of calling `SetConfig()` (which could have failed), we now call the assignment operator, which can't fail
4. We also validate once (and in one thread), so that's a nice side benefit
5. In addition to unrepresentable state of un-configured object, we now rely solely on whether the `cluster_config` (now called `tl_cluster_config` as it's **t**hread-**l**ocal) is `nullptr` or not, so that's another impossible state that we eliminated (no need to `CHECK(cluster_config == nullptr || cluster_config->IsConfigured())`)

So.. a refactor, but one that I'm fairly happy with!

Reply:
It looks like keeping all data per thread got us a `bad_alloc` exception on ARM. I guess that with many processors, the data here can be quite big, and when multiplied by many processors, it can be in the megabytes.
I doubt this will be of any impact on real setups, but thinking about this, pretty much the only reason for the huge `slots_` array is to reply with `MOVED` quickly. If we opt to reply there with O(n) (where `n` is the number of nodes in the cluster, which is not expected to be a huge numbers), we can drop that array entirely.
wdyt?

Reply:
I pushed that change to both see if it fixes the presubmit check, and also for us to see what it would look like